src/engine/enginebuffer.h

```diff
@@ -90,6 +82,8 @@ class EngineBuffer : public EngineObject {
         KEYLOCK_ENGINE_COUNT,
     };
 
+    static constexpr double kInitalSample = DBL_MIN;
```

Comment:
```suggestion
    static constexpr double kInitalSamplePosition = DBL_MIN;
```

Reply:
Also, could you add a comment why we use `DBL_MIN` instead of 0.0 or -1.0?

Reply:
Oh I think I original meant to use to use -DBL_MAX. 
0.0 does not work, because a track without any cue point will cued at 0.0. 

I think the -1.0 we use all over the code for an invalid position sourced from the days where we could not seek before the track. Now we have to live with the limitation that we can't place cue points at -1.0. I don't wanted to repeat the issue here again nor deal with an extra flag for this purpose.    

I have now changed this to -DBL_MAX which is -1.79769e+308, representing a cue point way before the big bang. 
We may also consider to use -INF or NAN, but I am not sure if our code is aware of that